omod/src/main/java/org/openmrs/module/aijar/web/controller/AppointmentController.java

```diff
@@ -0,0 +1,54 @@
+
+package org.openmrs.module.aijar.web.controller;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+ @Controller
+//@RequestMapping("/rest/ugandaemr/appointments") 
+public class AppointmentController { 
+
+    AdministrationService administrationService = Context.getAdministrationService();
+
+    @RequestMapping(value = "/rest/ugandaemr/appointments", method = RequestMethod.GET)
+    @ResponseBody
+    public Integer getNumberOfPatients(
+            @RequestParam(required = true, value = "nextAppointmentDate") Date nextAppointmentDate,
+            @RequestParam(value = "encounterType", defaultValue = "") String encounterType) {
+
+                SimpleDateFormat  formatter = new SimpleDateFormat("yyyy-MM-dd"); 
+                String returnDate = formatter.format(nextAppointmentDate); 
+        List patients = new ArrayList();
+
+        if (StringUtils.isEmpty(encounterType)) {
+            String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='"
+                    + returnDate + "' AND obs.voided = 0";
+            patients = administrationService.executeSQL(sql, true);
+
+        } else {
+            String[] encounterT = encounterType.split(",");
+            for (String e : encounterT) {
+                String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='"
+                        + returnDate
+                        + "' AND obs.voided = 0 AND obs.encounter_id IN ( SELECT encounter.encounter_id FROM encounter INNER JOIN encounter_type ON encounter.encounter_type = encounter_type.encounter_type_id    WHERE encounter_type.uuid IN ('"
+                        + e + "'))";
+                List patientsinEncounter = administrationService.executeSQL(sql, true);
+                patients.addAll(patientsinEncounter);
+            }
+
+        }
+        return patients.size();
```

Comment:
@reagan Your response has to include:
- date: the date submitted as part of the query 
- appointmentcount: the number of appointments on that date
- encountertypelist: the list of encounter types for the appointments